lib/game_client/ogs/ogs_websocket_manager.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:uuid/uuid.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+import 'package:web_socket_channel/status.dart' as status;
+
+class OGSWebSocketManager {
+  WebSocketChannel? _channel;
+  final String serverUrl;
+  String? _deviceId;
+  
+  final ValueNotifier<bool> _connected = ValueNotifier(false);
+  final StreamController<Map<String, dynamic>> _messageController = 
+      StreamController<Map<String, dynamic>>.broadcast();
+  
+  Timer? _pingTimer;
+  Timer? _reconnectTimer;
+  int _reconnectAttempts = 0;
+  static const int maxReconnectAttempts = 10;
+  
+  // For latency tracking
+  double _latency = 0.0;
+  double _clockDrift = 0.0;
+  
+  OGSWebSocketManager({required this.serverUrl}) {
+    _generateDeviceId();
+  }
+  
+  ValueNotifier<bool> get connected => _connected;
+  Stream<Map<String, dynamic>> get messages => _messageController.stream;
+  
+  void _generateDeviceId() {
+    // Generate a proper UUID for device ID like OGS frontend
+    const uuid = Uuid();
+    _deviceId = uuid.v4();
+  }
+  
+  Future<void> connect() async {
+    if (_channel != null) {
+      await disconnect();
+    }
+    
+    try {
+      // Convert HTTP(S) URL to WebSocket URL
+      final wsUrl = serverUrl
+          .replaceFirst('https://', 'wss://')
+          .replaceFirst('http://', 'ws://');
+      
+      debugPrint('Connecting to OGS WebSocket: $wsUrl');
+      
+      _channel = WebSocketChannel.connect(Uri.parse(wsUrl));
+      
+      _channel!.stream.listen(
+        _handleMessage,
+        onError: _handleError,
+        onDone: _handleDisconnect,
+      );
+      
+      _connected.value = true;
+      _reconnectAttempts = 0;
+      
+      // Start ping timer
+      _startPingTimer();
+      
+      debugPrint('Connected to OGS WebSocket');
+      
+    } catch (e) {
+      debugPrint('Failed to connect to OGS WebSocket: $e');
+      _scheduleReconnect();
+    }
+  }
+  
+  Future<void> disconnect() async {
+    _pingTimer?.cancel();
+    _reconnectTimer?.cancel();
+    
+    if (_channel != null) {
+      await _channel!.sink.close(status.goingAway);
+      _channel = null;
+    }
+    
+    _connected.value = false;
+  }
+  
+  Future<void> authenticate({required String jwtToken}) async {
+    if (_channel != null && _connected.value) {
+      final authData = {
+        'jwt': jwtToken,
+        'device_id': _deviceId,
+        'user_agent': 'WeiqiHub/1.0',
```

Comment:
maybe let's make a constant for the user agent, since it's used in a bunch of different places.